extensions/amp-carousel/0.1/slides.js

```diff
@@ -81,6 +93,55 @@ export class AmpSlides extends BaseCarousel {
             });
       }
     }
+    this.tryAutoplay_(1, true);
+  }
+
+  /**
+   * Sets up the `autoplay` configuration.
+   * @private
+   */
+  setupAutoplay_() {
+    if (!this.isAutoplaying_) {
+      return;
+    }
+
+    let autoplayValue = Number(this.element.getAttribute('autoplay'));
+    // If it isn't a number and is not greater than 0 then don't assign
+    // and use the default.
+    if (autoplayValue > 0) {
+      // Guard against autoplayValue that is lower than 1s to prevent
+      // people from crashing the runtime with providing very low delays.
+      this.autoplayDelay_ = Math.max(1000, autoplayValue);
+    }
+
+    // By default `autoplay` should also mean that the current carousel slide
+    // is looping. (to be able to advance past the last item)
+    if (!this.element.hasAttribute('loop')) {
+      this.element.setAttribute('loop', '');
+    }
+
+    // Setup microtask to let the runtime settle.
+    // TODO: (erwinm) experiment with only letting autoplay run if the
```

Comment:
yeah i should get to it tonight. issue => https://github.com/ampproject/amphtml/issues/605